packages/ui/src/components/TextField/TextField.tsx

```diff
@@ -167,7 +169,13 @@ export const TextField = (props: TextFieldProps) => {
 
   const [_value, setValue] = React.useState<Value>(value);
   const theme = useTheme();
-  const fallbackId = React.useId();
+
+  const {
+    errorScrollClassName,
+    errorTextId,
+    helperTextId,
+    validInputId,
+  } = useFieldIds({ errorGroup, hasError: Boolean(errorText), inputId, label });
 
   React.useEffect(() => {
     setValue(value);
```

Comment:
```suggestion
    if (value !== undefined) {
      setValue(value);
    }
```

I noticed the following console error when the input value changes in Storybook (but not in prod). I think we can resolve this issue by updating the value only if it's explicitly defined, and by adding a fallback above

![Screenshot 2025-04-01 at 11 21 42 AM](https://github.com/user-attachments/assets/bdcfcedc-e4e0-45bf-907d-a32c537c774d)